src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,267 @@
+            if need_replace.trim() == "y" {
+                fs::remove_file(&toolchain_info_path).unwrap();
+            } else {
+                bail!(
+                    "Failed to create a new toolchain info file at {} \
+                    because a toolchain info file already exists at that location.",
+                    &toolchain_info_path.display(),
+                );
+            }
+        }
+    }
+    let mut toolchain_name = Toolchain::from_settings()?.name;
+    if let Some(name) = name {
+        toolchain_name = name;
+    }
+
+    let export_toolchain = match DistToolchainDescription::from_str(&toolchain_name) {
+        Ok(desc) => Toolchain::from_path(&desc.to_string()),
+        Err(_) => Toolchain::from_path(&toolchain_name),
+    };
+
+    let mut version_map: HashMap<String, Version> = HashMap::new();
+    for component in Components::collect_exclude_plugins()? {
+        let component_executable = export_toolchain.bin_path.join(&component.name);
+        if let Ok(version) = exec_version(component_executable.as_path()) {
+            version_map.insert(component.name.clone(), version);
+        };
+
+        if component.name == component::FORC {
+            let forc_executables = component.executables;
+            for plugin in Components::collect_plugins()? {
+                if !forc_executables.contains(&plugin.name) {
+                    if !plugin.is_main_executable() {
+                        for executable in plugin.executables.iter() {
+                            let plugin_executable = export_toolchain.bin_path.join(executable);
+                            if let Ok(version) = exec_version(plugin_executable.as_path()) {
+                                version_map.insert(executable.clone(), version);
+                            };
+                        }
+                    } else {
+                        let plugin_executable = export_toolchain.bin_path.join(&plugin.name);
+                        if let Ok(version) = exec_version(plugin_executable.as_path()) {
+                            version_map.insert(plugin.name.clone(), version);
+                        };
+                    }
+                }
+            }
+        }
+    }
+    if toolchain_override::Channel::from_str(&toolchain_name).is_err() {
+        println_warning(&format!(
+            "Invalid channel '{}', expected one of <latest-YYYY-MM-DD|nightly-YYYY-MM-DD|beta-1|beta-2|beta-3|beta-4>. \
+            Please input a valid channel: ",
+            toolchain_name,
+        ));
+        let mut input_toolchain_name = String::new();
+        reader.read_line(&mut input_toolchain_name).unwrap();
+        input_toolchain_name = String::from(input_toolchain_name.trim());
+        if toolchain_override::Channel::from_str(&input_toolchain_name).is_err() {
+            bail!(
+                "Invalid channel '{}', expected one of <latest-YYYY-MM-DD|nightly-YYYY-MM-DD|beta-1|beta-2|beta-3|beta-4>.",
```

Comment:
In the interest of DRY, let's move this to a constant

```
const INVALID_CHANNEL_STR: &str = "Invalid channel '{}', expected one of <latest-YYYY-MM-DD|nightly-YYYY-MM-DD|beta-1|beta-2|beta-3|beta-4>.";
```

Reply:
Done